tests/unit_tests/service/test_authentication.py

```diff
@@ -132,3 +129,12 @@ def test_processes_valid_token(
 ):
     inner = main.verify_access_token(oidc_config)
     inner(access_token=valid_token_with_jwt["access_token"])
+
+
+def test_session_cache_manager_returns_writable_file_path(tmp_path):
+    with patch("os.environ.get") as xdg_directory:
+        xdg_directory.return_value = tmp_path
+        cache = SessionCacheManager(token_path=None)
+        with open(cache._file_path, "xb") as token_file:
+            assert token_file.writable()
```

Comment:
Can this ever be false? If the path isn't writable wouldn't the `open` call fail?

Reply:
This can be `False` If the user does not have permission to write the file at the desired location

Reply:
If it's not writable, isn't that a problem with the `tmp_path` fixture and not the session cache?

Reply:
The purpose of this test is to ensure that `SessionCacheManager` always returns a file path rather than a directory, allowing BlueAPI to write the `token_cache` to the specified `file_path`. While the `tmp_path` fixture itself shouldn't cause issues, this approach validates the behavior by touching the file, ensuring the path is writable. 

Currently, I’m not aware of a  way to confirm if a path is specifically a file path and not a directory without either opening or attempting to touch the file. If there’s a more elegant solution to this, I’d be open to suggestions.

I have changed the test slightly since last review ..Instead of open the file ,I'm doing `touch` on the path provide and then doing `isFile` on the given path...The `touch` should raise a error if Session_cache returns a directory path rather than a file path